core/src/main/java/gov/nist/secauto/metaschema/core/metapath/function/library/FnEndsWith.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * SPDX-FileCopyrightText: none
+ * SPDX-License-Identifier: CC0-1.0
+ */
+
+package gov.nist.secauto.metaschema.core.metapath.function.library;
+
+import gov.nist.secauto.metaschema.core.metapath.DynamicContext;
+import gov.nist.secauto.metaschema.core.metapath.ISequence;
+import gov.nist.secauto.metaschema.core.metapath.MetapathConstants;
+import gov.nist.secauto.metaschema.core.metapath.function.FunctionUtils;
+import gov.nist.secauto.metaschema.core.metapath.function.IArgument;
+import gov.nist.secauto.metaschema.core.metapath.function.IFunction;
+import gov.nist.secauto.metaschema.core.metapath.item.IItem;
+import gov.nist.secauto.metaschema.core.metapath.item.atomic.IBooleanItem;
+import gov.nist.secauto.metaschema.core.metapath.item.atomic.IStringItem;
+
+import java.util.List;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+public final class FnEndsWith {
+
+  @NonNull
+  static final IFunction SIGNATURE = IFunction.builder()
+      .name("ends-with")
+      .namespace(MetapathConstants.NS_METAPATH_FUNCTIONS)
+      .deterministic()
+      .contextDependent()
+      .focusIndependent()
+      .argument(IArgument.builder()
+          .name("arg1").type(IStringItem.class)
+          .zeroOrOne()
+          .build())
+      .argument(IArgument.builder()
+          .name("arg2")
+          .type(IStringItem.class)
+          .zeroOrOne()
+          .build())
+      .returnType(IBooleanItem.class)
+      .returnOne()
+      .functionHandler(FnEndsWith::execute)
+      .build();
+
+  @SuppressWarnings("unused")
+  @NonNull
+  private static ISequence<IBooleanItem> execute(@NonNull IFunction function,
+      @NonNull List<ISequence<?>> arguments,
+      @NonNull DynamicContext dynamicContext,
+      IItem focus) {
+    IStringItem arg1 = FunctionUtils.asTypeOrNull(arguments.get(0).getFirstItem(true));
+
+    IStringItem arg2 = FunctionUtils.asTypeOrNull(arguments.get(1).getFirstItem(true));
+
+    return ISequence.of(IBooleanItem.valueOf(
+        fnEndsWith(
+            arg1 == null ? "" : arg1.asString(),
+            arg2 == null ? "" : arg2.asString())));
+  }
```

Comment:
Non-blocking, but important: per [the documentation](https://www.w3.org/TR/xpath-functions-31/#func-ends-with), we are not going to implement the three argument form? I am asking, I know that the collation handling may be juice not worth the squeeze, but you get me reading official docs a lot now, so I thought I may as well as ask.

Reply:
See #170